schema/output_parsers.go

```diff
@@ -0,0 +1,10 @@
+package schema
+
+type OutputParser[O any] interface {
```

Comment:
This would be the first introduction of generics to the codebase(to my knowledge). 

Can you sketch out how we’d approach this without them as well as with them?

Reply:
TBH, I am not sure about this for now, I am still trying to dig into the source code and find out the workaround to avoid generics

Reply:
let's do it but call the type arg "T"

Reply:
Ok, I will update it later.